File: BookLocal.API/DTOs/ServiceCategoryDto.cs
using System.ComponentModel.DataAnnotations;

namespace BookLocal.API.DTOs
{
    public class ServiceCategoryUpsertDto
    {
        [Required]
        [MaxLength(100)]
        public required string Name { get; set; }
        [Required]
        public int MainCategoryId { get; set; }
    }
    public class ServiceCategoryDto
    {
        public int ServiceCategoryId { get; set; }
        public required string Name { get; set; }
        public string? PhotoUrl { get; set; }
        public List<ServiceDto> Services { get; set; } = new();
    }

    public class ServiceCategoryFeedDto
    {
        public int ServiceCategoryId { get; set; }
        public required string Name { get; set; }
        public string? PhotoUrl { get; set; }
        public int BusinessId { get; set; }
        public required string BusinessName { get; set; }
        public string? BusinessCity { get; set; }
        public IEnumerable<ServiceDto> Services { get; set; } = new List<ServiceDto>();
    }
}

File: booklocal_mobile/lib/core/models/service_category_dto.dart
import 'service.dart';

class ServiceCategoryDto {
  final int serviceCategoryId;
  final String name;
  final String? photoUrl;
  final bool isArchived;
  final List<Service> services;

  ServiceCategoryDto({
    required this.serviceCategoryId,
    required this.name,
    this.photoUrl,
    required this.isArchived,
    required this.services,
  });

  factory ServiceCategoryDto.fromJson(Map<String, dynamic> json) {
    return ServiceCategoryDto(
      serviceCategoryId: json['serviceCategoryId'] ?? 0,
      name: json['name'] ?? '',
      photoUrl: json['photoUrl'],
      isArchived: json['isArchived'] ?? false,
      services: (json['services'] as List<dynamic>?)
              ?.map((s) => Service.fromJson(s))
              .toList() ??
          [],
    );
  }
}
